### PR TITLE
Resprites power cells, fixes and cleans up power cells and power cell derivatives

### LINKED ISSUE #62200: Cells do not update their sprites on chargers
This is actually a really old bug but it'd be really nice if this was fixed, Cells have varying sprites for their different charge levels but they stick with the uncharged sprite until you take them off the dock.

### PR DESCRIPTION
**Pull request by Thunder12345**
## About The Pull Request

Resprited power cells and the cell charger in 3/4th style

Cleaned up a number of bug/oversights related to power cells

Removed an uncraftable 15k charge cell variety that was present in numerous places and replaced it with the 10k cell

image

Fixes #62200

Closes #64899

## Why It's Good For The Game

Modernises power cell and charger sprites and fixes a selection of power cell related bugs/oversights.

Removal of the 15k cell cleans up an uncraftable variety of power cell that doesn't serve any particular purpose, removal done with the approval of Armhulen. This change is unlikely to have serious balance implications, as it was only present loose on maps, and as a replaceable part of prebuilt borg shells, mechs, and some modsuits, and superior supercaps are generally available early on in the round.

image

## Changelog

:cl:
imageadd: Power cells and the cell charger have been resprited, the charger will now look correct with a much wider variety of non-standard power cells
del: Removed the uncraftable 15k charge high-capacity power cell+, all instances have been replaced with the regular 10k charge high-capacity cell
fix: Erroneous power cell charge indicators have been removed from many non-standard types of power cells
fix: All varieties of marauder mech including the nuke ops mauler come with a bluespace power cell, consistent with other nuke op mechs
fix: Power cell charge indicators will now update while in a cell charger
fix: Potato and other crop batteries will not have their wires disappear when on cell chargers
/:cl:

### CODE REVIEW

**Comment by Kubisopplay** [comment]
Thunder, the 15k cell is for borgs, so they can get a small treat from robos. Could you replace them with 20k then?

**Comment by Serxule** [comment]
can we see the sprites?

**Comment by Thunder12345** [comment]
> Thunder, the 15k cell is for borgs, so they can get a small treat from robos. Could you replace them with 20k then?

There was never any treat, they started with 15ks already. Be patient for like 10 minutes then go pick up your supercap

**Comment by cacogen** [comment]
Armhulen just dived into the conversation and gave his kneejerk reaction to one thing you wrote so he could then proceed to make it about something else so I'm not sure if his approval is coming from a very informed place.

**Comment by Thunder12345** [comment]
4452F672-BF3F-469F-8B80-1F1659D4ABE3

When I suggested keeping them in specifically for borg etc applications Arm said he didn’t see the point of keeping them for a single purpose, so it’s not like he appeared once to say two words then never looked at the conversation again

**Comment by zxaber** [comment]
> There was never any treat, they started with 15ks already.

Borgs start with 10k cells.

It's nice to be able to give the shift-start borg a small power upgrade if they come and visit early on.

**Comment by Thunder12345** [comment]
> > There was never any treat, they started with 15ks already.
> 
> Borgs start with 10k cells.
> 
> It's nice to be able to give the shift-start borg a small power upgrade if they come and visit early on.

No, before this pr they started with 15ks

**Comment by Bawhoppen** [comment]
The colors feel a little washed out on the sprites. Maybe look at increasing contrast and brightness?

**Comment by optimumtact** [comment]
from memory those plus cells were so robots were motivated to visit the roboticist early on.

taht's their only purpose and why you can't make more of htem.

Of course it was all fucked up later when borgs got better batts at roundstart anyway.

**Comment by tralezab** [comment]
> Armhulen just dived into the conversation and gave his kneejerk reaction to one thing you wrote so he could then proceed to make it about something else so I'm not sure if his approval is coming from a very informed place.

I have double checked and yes they should be on their way out

**Comment by cacogen** [comment]
The explanation for why they were added is here: https://github.com/tgstation/tgstation/pull/12434#issuecomment-148877859

**Comment by zxaber** [comment]
> No, before this pr they started with 15ks

In robot_defines.dm;
`var/obj/item/stock_parts/cell/cell = /obj/item/stock_parts/cell/high`
In robot.dm, under `Initialize()`;
```
if(ispath(cell))
		cell = new cell(src)
```

I'm willing to accept I'm wrong, but the code *seems* to suggest they get 10k highcaps. Have you checked in-game on a round-start borg?

**Comment by Thunder12345** [comment]
Nevermind you’re right, that was prebuilt borg shells not actual borgs that had the 15k

**Comment by tralezab** [comment]
So considering they weren't getting 15k cells can we agree that the cells going is a nothingburger

**Comment by Imaginos16** [comment]
lol

**Comment by Thunder12345** [comment]
> The explanation for why they were added is here: [#12434 (comment)](https://github.com/tgstation/tgstation/pull/12434#issuecomment-148877859). Then you have the God King Kor Himself ordaining that a 15k power cell is to be created.

Wake up you’ve been in a coma for four years, Kor retired to the emu farm while you were sick

**Review by Imaginos16 — APPROVED** [review]

**Comment by cacogen** [comment]
Why don't you respond to the rest of that comment?

File: code/modules/power/apc.dm [hunk 1]
```diff
@@ -184,9 +184,6 @@
 /obj/machinery/power/apc/highcap/ten_k
```
Comment (Fikou) [inline code/modules/power/apc.dm:184]: the highcap type doesnt exist, remove this subtype and make it the highcap type

File: code/modules/mod/mod_types.dm [hunk 2]
```diff
@@ -186,7 +184,6 @@
 
 /obj/item/mod/control/pre_equipped/prototype
 	theme = /datum/mod_theme/prototype
-	applied_cell = /obj/item/stock_parts/cell/high/plus
```
Comment (Fikou) [inline code/modules/mod/mod_types.dm:189]: make this the super capacity power cell

**Comment by cacogen** [comment]
All these things having to be either nerfed or buffed to compensate for a change that didn't need to be made in the first place.

File: code/modules/power/apc.dm (hunk 1 above)
Comment (Thunder12345) [reply code/modules/power/apc.dm:184]: The highcap type itself doesn’t exist, but there’s still also the 5k version

**Comment by Serxule** [comment]
a corner being cut off does not make any sense, but sprites are fine otherwise

**Comment by Serxule** [comment]
actually, the sprites arent very good at all, they all look identical and i could only tell you which one is which due to it being laid out, whereas the old sprites can actually be identified due to certain markings.

**Comment by tuckerorchistra** [comment]
these look like fucking shit, if anything it looks like a fucking charger more than a battery AND how is that shit supposed to fit into my laser pointer

**Comment by zxaber** [comment]
I like the shape of them, but the color isn't as straightfoward as before.  This all loses the neat one/two/three mark system that we had before for tiers 1-3. It also implies that Hypers are an improvement within the T2 category, rather than being their own tier altogether.

They do look nice, visually, but it doesn't easily convey their hierarchy well. Maybe that's not really important, though; there's only four main ones you'll see and that's not a hard amount to remember.

**Comment by cacogen** [comment]
The fixes and sprites should be atomised into two different PRs rather than attaching desirable changes to a self-serving, divisive one. 

I like the sprites but the colour schemes should reflect what we've always had and you should be able to differentiate the different tiers of cell at a glance the way you could with the stripes on top of the old ones.

**Comment by Donpedrito** [comment]
> these look like fucking shit, if anything it looks like a fucking charger more than a battery AND how is that shit supposed to fit into my laser pointer

laser pointers don't have batteries, they have micro-lasers

**Comment by MetalClone** [comment]
Quaint idea, must include the abductor alien battery though for consistency

**Comment by timothymtorres** [comment]
@Thunder12345 does this fix https://github.com/tgstation/tgstation/issues/62182 or https://github.com/tgstation/tgstation/issues/48259

**Comment by optimumtact** [comment]
> these look like fucking shit, if anything it looks like a fucking charger more than a battery AND how is that shit supposed to fit into my laser pointer

@tuckerorchistra  If you're unable to offer any actual guideline/advice to what looks incorrect or bad about them you will be banned from this github, i'm not interested in criticism that is surface level and rude.

**Pull request merged**